architecture/2024-08-resource-types-registration.md

```diff
@@ -0,0 +1,580 @@
+# Resource Type Registration APIs
+
+* **Author**: `Ryan Nowak` (`@rynowak`)
+
+## Overview
+
+*This document describes the registration APIs resource types. This is a detailed design doc focused on one aspect of the overall user-defined types feature area. Please see the overall user-defined-types plan for more context.*
+
+Resources are the unit of exchange in the Radius API. All of our user-experiences like the CLI, Dashboard, and Bicep authoring are oriented around the catalog of resources. Because Radius is strongly-typed, the set of resource types and their schemas are also defined as a primary concept in Radius. In order for Radius to support each additional kind of resources, for example a PostgreSQL database, we must first have a type definition.
+
+The registration APIS for resource types determine how users add, change, or remove types from Radius.
+
+## Terms and definitions
+
+*Please read the [Radius API](https://docs.radapp.io/concepts/technical/api/) conceptual documentation. This document will heavily use the terminology and concepts defined there.*
+
+*Please see additional terms and context in the [overall user-defined-types](https://github.com/radius-project/design-notes/pull/56/files) design note.*
+
+| Term               | Definition                                                                                                                                                                                                                                                                                   |
+| ------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Resource Namespace | A group of related or similar resources that share a resource provider. eg: In `Applications.Core/containers` the resource namespace is `Applications.Core`.                                                                                                                                 |
+| Resource Provider  | A resource namespace and implementation of the resource namespace. Resource providers are a new resource type introduced by this proposal.                                                                                                                                                   |
```

Comment:
What is an example of this? Like the whole process that runs to serve to the requests for this resource namespace?

Reply:
An example of this would be `Applications.Core` + `applications-rp` in Radius today. It's both the set of supported types and the implementation.